### PR TITLE
Require dependencies script

### PR DESCRIPTION
**Pull request by Proximyst**
This adds a new script which ensures all required dependencies are present.

This also adds a dependency I missed into the `CONTRIBUTING.md` file.

### CODE REVIEW

**Comment by MrIvanPlays** [comment]
I think wiggle should be optional; it builds without it.

**Comment by Prof-Bloodstone** [comment]
I agree that `wiggle` is optional, and shouldn't be required to run the script most of the time, but when it gets required, I think it'd make sense to check for it: https://github.com/PaperMC/Paper/blob/86daffa21c13f46361313120dd49f4bb9118b8d9/scripts/apatch.sh#L32

Idk whether it should be checked and abort due to it being required at that stage, or maybe that stage should be skipped with a message like:
> Unable to wiggle patches automatically - install 'wiggle' or apply them manually

What do you think?

**Comment by electronicboy** [comment]
apatch isn't really part of the paper scripts, it's just a tool we keep in there to share as it can help with stubborn merge conflicts

**Comment by Proximyst** [comment]
As `apatch.sh` is an internal script, I'll opt to not add a requirement to it. If you're using that script, you should know how to diagnose its issues.

File: scripts/requireDeps.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
```
Comment (Prof-Bloodstone) [inline scripts/requireDeps.sh:1]:
```suggestion
#!/usr/bin/env sh
```

**Review by Prof-Bloodstone — DISMISSED** [review]
That makes sense. LGTM

File: scripts/requireDeps.sh (hunk 1 above)
Comment (Proximyst) [reply scripts/requireDeps.sh:1]: POSIX defines that `/bin/sh` must be a shell, always. No need to use env for this.

File: scripts/requireDeps.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+if [ -z "${1:-}" ]; then
+    # No specific dependency was found; let's just check for all required ones.
+    _is_dep_available git
+    _is_dep_available patch
+    _is_dep_available mvn
+
+    # Ensure we don't have a JAVA_HOME set first.
+    # Maven should work fine without the JAVA_HOME var as long as the JDK is on the PATH.
+    if [ -z "${JAVA_HOME:-}" ]; then
+        _is_dep_available javac "was not found; you can download the JDK from https://adoptopenjdk.net/ or your package manager"
```
Comment (kashike) [inline scripts/requireDeps.sh]:
```suggestion
        _is_dep_available javac "was not found; you can download the JDK from https://adoptopenjdk.net/ or via your package manager"
```

**Review by kashike — DISMISSED** [review]
Looks good to me; one comment.

**Review by kashike — APPROVED** [review]

**Pull request merged**